### PR TITLE
Get process list after PID 1 dead

### LINKED ISSUE #11087: Orphaned and unstoppable (superhuman strength?) containers with --pid=host
I've ran into an issue that is easily reproducible that puts a container into a state that can not be stopped and if you restart the Docker daemon it will also loose track of the existing container and spawn a new one.

The issue happens when you are running with `--pid=host` and you do a SIGKILL but that process has children that don't terminate when its parent dies.  Docker never seems to see that the SIGKILL worked and will hang forever.  The SIGKILL worked, but the container PID is gone.  Now you do a `docker stop` or any action and Docker will complain that it can't find the process.

Now if you restart the Docker daemon, Docker thinks the container is gone (I assume because the PID is gone) and then it will start a new one but the old container cgroup still exists.

I haven't dug yet into the Docker code to see what is happening (I assume Docker thinks the SIGKILL isn't done because processes in the cgroup still exist), but here is a simple way to reproduce this issue.

```bash
#!/bin/bash

ID=$(docker run -d --pid=host ubuntu bash -c "sleep 300; echo hi")

docker kill $ID &
sleep 1
docker stop $ID
```

Running the above will result in

```
Error response from daemon: Cannot stop container 8af7796dff4df03686342f3a7b618fe5f1830ab89c0c158817acb332cac3e9f8: no such process
FATA[0000] Error: failed to stop one or more containers 
```

Seems like there are two things that should be addressed.

1. Make SIGKILL work properly in this situation
2. On restart check the cgroup for processes and freeze/kill those

Obligatory docker info/version

```
$ docker version; docker info
Client version: 1.5.0
Client API version: 1.17
Go version (client): go1.4.1
Git commit (client): a8a31ef
OS/Arch (client): linux/amd64
Server version: 1.5.0
Server API version: 1.17
Go version (server): go1.4.1
Git commit (server): a8a31ef
Containers: 51
Images: 366
Storage Driver: overlay
 Backing Filesystem: extfs
Execution Driver: native-0.2
Kernel Version: 3.18.1-031801-generic
Operating System: Ubuntu 14.10
CPUs: 4
Total Memory: 7.496 GiB
Name: inotmac
ID: FPSO:Z7SO:BZJ3:IXH4:DNKV:UPZQ:KLP2:IRXW:5CBY:PBDM:HAKO:UOC7
Username: ibuildthecloud
Registry: [https://index.docker.io/v1/]
```

### PR DESCRIPTION
**Pull request by LK4D4**
Fix #11087

### CODE REVIEW

**Comment by jfrazelle** [comment]
nice! and idk how that other got removed from milestone... want to blame waffle!

**Comment by jfrazelle** [comment]
ping @ibuildthecloud if you wanna test

**Comment by jfrazelle** [comment]
LGTM

**Comment by ibuildthecloud** [comment]
I'll test tonight, traveling atm.  The branch name is quite fitting :)

**Comment by jfrazelle** [comment]
lol just noticed that

File: integration-cli/docker_cli_run_test.go [hunk 1]
```diff
@@ -3468,3 +3468,27 @@ func TestRunContainerWithRmFlagCannotStartContainer(t *testing.T) {
 
 	logDone("run - container is removed if run with --rm and cannot start")
 }
+
+func TestRunPidHostWithChildIsKillable(t *testing.T) {
+	defer deleteAllContainers()
+	name := "ibuildthecloud"
```
Comment (jfrazelle) [inline integration-cli/docker_cli_run_test.go]: ah a dedication

**Comment by jfrazelle** [comment]
test passes on master is this expected?

File: integration-cli/docker_cli_run_test.go [hunk 2]
```diff
@@ -3468,3 +3468,27 @@ func TestRunContainerWithRmFlagCannotStartContainer(t *testing.T) {
...
+	if out, err := exec.Command(dockerBinary, "run", "-d", "--name", name, "busybox", "sh", "-c", "sleep 300; echo hi").CombinedOutput(); err != nil {
```
Comment (tiborvass) [inline integration-cli/docker_cli_run_test.go]: no need for that long of a sleep, and no need for `echo hi` either.

Comment (tiborvass) [inline integration-cli/docker_cli_run_test.go]: `"busybox", "sleep", "30")` should be enough

Comment (jfrazelle) [inline integration-cli/docker_cli_run_test.go]: I think it was just an exact duplication of https://github.com/docker/docker/issues/11087

File: daemon/execdriver/native/driver.go [hunk 3]
```diff
@@ -206,10 +204,12 @@ func waitInPIDHost(p *libcontainer.Process, c libcontainer.Container) func() (*o
 			}
 			s = execErr.ProcessState
 		}
+
+		processes, err := c.Processes()
```
Comment (crosbymichael) [inline daemon/execdriver/native/driver.go]: Shouldn't you be freezing the container, getting the pids, sending the signals, then unfreeze and wait on the individual processes like the code does in container.Destroy()?

**Comment by LK4D4** [comment]
@jfrazelle Yup, we need little delay between run and kill to let child spawn.

**Comment by LK4D4** [comment]
@crosbymichael I did as you said.
NOTE: test is passing on master
Bug is easy reproducible with:
```
ID=$(docker run -d --pid=host busybox sh -c "top") && sleep 1 && docker kill $ID
```
But not reproducible inside testing container.
I don't know what to do, I can just remove test. Let me know.

**Comment by jfrazelle** [comment]
I dont know I mean a test is better than nothing, who knows maybe something will break it eventually lol

**Comment by crosbymichael** [comment]
idk about the test, i tested locally and it works.  it's a hard test because docker has to be PID 1

**Comment by crosbymichael** [comment]
LGTM

**Comment by ehazlett** [comment]
LGTM

**Pull request merged**